python-client/giskard/integrations/mlflow/giskard_evaluator_utils.py

```diff
@@ -0,0 +1,126 @@
+import inspect
+import re
+import logging
+
+import langchain.chains.base
+
+from giskard import Model, Dataset, scan
+from giskard.core.core import SupportedModelTypes
+from giskard.core.model_validation import ValidationFlags
+
+logger = logging.getLogger(__name__)
+
+gsk_model_types = {
+    "classifier": SupportedModelTypes.CLASSIFICATION,
+    "regressor": SupportedModelTypes.REGRESSION,
+    "text": SupportedModelTypes.TEXT_GENERATION,
+}
+alphanumeric_map = {
+    ">=": "greater than or equal to",
+    ">": "greater than",
+    "<=": "less than or equal to",
+    "<": "less than",
+    "==": "equal to",
+    "=": "equal to",
+    "!=": "different of",
+}
+
+
+def unwrap_python_model_from_pyfunc(pyfunc_model):
+    """
+    An alternative to https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=pyfunc#mlflow.pyfunc.PyFuncModel.unwrap_python_model
+    :param pyfunc_model: the pyfunc model
+    :return: unwrapped model
+    """
+    import importlib
+
+    module = pyfunc_model.metadata.flavors["python_function"]["loader_module"]
+    flavor = importlib.import_module(module)
+    return flavor.load_model(pyfunc_model.metadata.get_model_info().model_uri)
+
+
+def process_text(some_string):
+    for k, v in alphanumeric_map.items():
+        some_string = some_string.replace(k, v)
+    some_string = some_string.replace("data slice", "data slice -")
+    some_string = re.sub(r"[^A-Za-z0-9_\-. /]+", "", some_string)
+
+    return some_string
+
+
+def setup_dataset(dataset, evaluator_config):
+    data = dataset.features_data.copy()
+    target = dataset.targets_name if hasattr(dataset, "targets_name") else None
+    if target:
```

Comment:
You should check for `None` because target could in principle be numerical `0`, which would evaluate to `False`.